make event colors optional

etwillbefine opened the pull request:
/cc @mattlewis92 

relates to https://github.com/mattlewis92/angular-calendar/issues/468

etwillbefine commented:
Not sure about the `defaults.scss`. Couldnt find any other suitable place. 

ejs was missing for me. Installed the module but now the package-lock went crazy. Shall I remove the commit again?

mattlewis92 commented on src/defaults.scss:
```diff
@@ -0,0 +1,6 @@
+
+.cal-event {
```
Rather than having `defaults.scss` I think it's better to just have the CSS copied under each of the views, it also prevents adding the global `cal-event` rule to the whole app

mattlewis92 replied:
Also for the month view the background color is the primary color, and for the day / week views its the secondary color, and the border color is the primary one

etwillbefine replied:
I understand. But wouldn't it be easier then, to override the cal-event once? Just to be sure, for me it doesn't matter but imo it makes it less "complicated"

mattlewis92 commented on src/defaults.scss:
```diff
@@ -0,0 +1,6 @@
+.cal-event {
+  background-color: #0062cc;
+  border-color: #005cbf;
+  color: #fff;
```
How come this is needed?

etwillbefine replied:
If there is nothing supplied it would use the primary color instead the one which fits to the background. That was my thought.

etwillbefine replied:
If there is no color specified it will use the font set on a parent. It may doesn't fit to the background and needs extra Styling. That was my thought.

mattlewis92 commented on package.json:
```diff
@@ -124,6 +124,7 @@
     "core-js": "^2.5.1",
     "create-plunker": "^1.4.0",
     "css-loader": "^0.28.9",
+    "ejs": "^2.5.7",
```
This shouldn't be needed - are you running the latest version of npm? If you copy the package-lock.json from master and `rm -rf node_modules` first before doing `npm i` it should do the trick.

etwillbefine replied:
I checked out the project from scratch and did npm I. When running the tests it complained that ejs was missing. I'll give it another try and update my npm to latest.

mattlewis92 commented on test/calendar-week-view.component.spec.ts:
```diff
@@ -92,6 +92,37 @@ describe('calendarWeekView component', () => {
+      fixture.nativeElement.querySelector('.cal-event')
+    );
+    expect(computedStyles.getPropertyValue('background-color')).to.equal(
+      'rgb(0, 98, 204)'
```
That's a clever way of testing this change, nice idea! 😄

etwillbefine commented on src/modules/day/calendar-day-view.scss:
```diff
@@ -48,14 +48,20 @@
+  }
+
+  .cal-event-title:link {
+    color: currentColor;
```
sorry for this additional change. I can also remove it again but it was strange to see dots from ellipsis in the specified color and the link in blue (bootstrap) in the demos..

mattlewis92 replied:
I learnt something new today from this, thanks! 😄

etwillbefine commented:
I changed the colors to use the existing blue scheme from the demos. Bootstrap was using grey as secondary color... Hope that this addresses all your concerns.

etwillbefine commented:
Thanks for your hints! The event-title color is already white, it was the link color again. 
Imo a central place for common styles is missing. Needed to duplicate some code now (for the link color) ..

Border-color on the week view will now update according to the color specified in the event. 
I'm on vacation from tonight. Might take a little longer to respond.

mattlewis92 commented:
Perfect, thanks a lot Tim! Have an awesome easter weekend! 😄

mattlewis92 commented:
Whoops, sorry I completely forgot to release this, done now in `0.24.0` 😄
